perception/realsense_sensor.py

```diff
@@ -0,0 +1,241 @@
+"""
+Class for interfacing with the Intel RealSense D400-Series.
+"""
+import logging
+import numpy as np
+
+try:
+    import pyrealsense2 as rs
+except ImportError:
+    logging.warning('Unable to import pyrealsense2.')
+
+from perception import CameraIntrinsics, CameraSensor, ColorImage, DepthImage
+
+try:
+    import rospy
+    from rospy import numpy_msg
+except ImportError:
+    logging.warning("Failed to import ROS in realsense_sensor.py. ROS functionality not available")
+try:
+    from perception.srv import *
+    ImageBufferResponse = rospy.numpy_msg.numpy_msg(ImageBufferResponse)
+    ImageBuffer._response_class = ImageBufferResponse
+except ImportError:
+    logging.warning('realsense_sensor.py not installed as catkin package. ROS functionality not available.')
+
+
+class RealSenseRegistrationMode:
+    """Realsense registration mode.
+    """
+    NONE = 0
+    DEPTH_TO_COLOR = 1
+
+
+class RealSenseSensor(CameraSensor):
+    """Class for interacting with a RealSense D400-series sensor.
+    """
```

Comment:
Also, pip binaries are not available for all architectures (see this [issue](https://github.com/IntelRealSense/librealsense/issues/2586)). I do however have instructions for compiling from source (I work with macOS). How would you like me to proceed?

Reply:
It would be great if you can include those instructions (including mac compatibility) right where I left the comment. That way the next time we compile the docs, it will be there in the RealSenseSensor reference.

Also, if the mac install instructions are long, then feel free to leave a link to a Google doc or some other public instructions.

Thanks!

Reply:
@jeffmahler Done! I also added a test script in `tests/`.

Reply:
Looks great!